lizmap/modules/admin_api/lib/RepoCreator.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace LizmapApi;
+
+class RepoCreator
+{
+    /**
+     * Creates a new repository with the given properties.
+     *
+     * @param string $key                    The unique identifier for the repository. Must not yet exist.
+     * @param string $label                  the display name of the repository
+     * @param string $path                   the path to the repository directory
+     * @param string $allowUserDefinedThemes whether user-defined themes are allowed for the repository
+     *
+     * @return bool returns true if the repository is successfully created
+     *
+     * @throws \Exception if the repository creation fails
+     */
+    public static function createRepository(string $key, string $label, string $path, string $allowUserDefinedThemes): bool
+    {
+        if (\lizmap::getRepository($key)) {
+            throw new \Exception("The repository '{$key}' already exists.", 409);
+        }
+
+        if (!$label) {
+            throw new \Exception('The repository label is not set.', 400);
+        }
+
+        if (!$path) {
+            throw new \Exception('The repository path is not set.', 400);
+        }
+
+        $data = array(
+            'label' => $label,
+            'path' => $path,
+            'allowUserDefinedThemes' => $allowUserDefinedThemes,
+        );
+
+        $rootRepositories = \lizmap::getServices()->getRootRepositories();
```

Comment:
There is a setting in LWC admin panel to choose or not if we have a root folder.